### ISSUE TITLE
Display semantic errors after overriding controller

### ISSUE THREAD

**Issue by usama520** [activeadmin/activeadmin#4826]
After overriding the create method in activeadmin, I want to display the validation errors on the fields (example: "Name can't be blank" or "name already exists"). I couldn't find an elegant solution and the only solution according to me is to know that how is activeadmin implementing this. Please guide me towards it.

Thanks in advance.

**Comment by timoschilling** [comment]
Why do you overwrite `create`? Show us how you overwrite it.

**Comment by usama520** [comment]
I had to override create because I had a situation where I had to autofill all the fields and after that if the data in the fields is already present, update the data otherwise create. The form is a nested form and I want to create the nested objects and display the validation errors if there are any. This is my create method

`    def create
      id = params[:company].dig(:id)
      if id.present?
        @company = Company.find(id)
        if @company.update(permitted_params[:company])
          redirect_to resource_url
          flash[:notice] = 'Company created successfully'
        else
          @errors = @company.errors
        end
      else
        new_permitted_params = permitted_params[:company].except(:id)
        @company = Company.new(new_permitted_params)
        @company.save
        if @company.errors.any?
           #add errors to semantic errors so that activeadmin displays it with the fields.
        end
`

**Comment by timoschilling** [comment]
We can't give support for such hacks. Take a look on inherited_resurce, active admin is based on it or ask on stackoverflow.

**Comment by usama520** [comment]
I just need to know how is activeadmin setting semantic errors. Once I know how to do that, my job is practically done.

**Comment by varyonic** [comment]
See the inherited_resources readme for suggestions how to override create.

**Comment by usama520** [comment]
@varyonic I dont think you understood the question. I didnt ask "how to override a controller". I asked how to display the semantic errors after overriding the controller. How are you guys setting the semantic errors in the form and displaying them? That's my question.

**Comment by Fivell** [comment]
@usama520 , https://github.com/activeadmin/activeadmin/blob/9541d9a33679d247c084ea87e62b63504aa79fda/lib/active_admin/views/pages/form.rb#L44

**Comment by varyonic** [comment]
Answered by Usama on [SO](http://stackoverflow.com/questions/42460672/how-to-display-errors-in-activeadmin-forms-after-overriding-controller)

**Comment by usama520** [comment]
@varyonic. Yes that's me. In the end I was able to find the solution. When I was creating an object with parameters violating the validations, the create method was looking for a template and couldn't find it. This was being displayed in the terminal. That's when it occurred to me to see which template was activeadmin using. I looked at it in the terminal by commenting the controller and that was my answer. Fivells's comment was technically of no help but I appreciate you all for trying for me.
